### PR TITLE
Implement organization creation permission controls

### PR DESCRIPTION
**Pull request by jescalan**
## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [x] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description
If the "user can create organizations" permission is false, the "create organization" button will not appear in the `OrganizationSwitcher` component.

### CODE REVIEW

File: packages/clerk-js/src/core/resources/User.ts [hunk 1]
```diff
@@ -80,6 +80,7 @@ export class User extends BaseResource implements UserResource {
   publicMetadata: UserPublicMetadata = {};
   unsafeMetadata: UserUnsafeMetadata = {};
   lastSignInAt: Date | null = null;
+  createOrganizationEnabled = false;
```
Comment (jescalan) [inline packages/clerk-js/src/core/resources/User.ts:83]: Should we actually default this to false?

Comment (nikosdouvlis) [reply packages/clerk-js/src/core/resources/User.ts:83]: It shouldn't matter in this case as a User is always initialized from json data.

You definitely have more context than me,but this seems fine!

**Review by nikosdouvlis — APPROVED** [review]

File: packages/types/src/userSettings.ts [hunk 2]
```diff
@@ -83,14 +83,14 @@ export type Attributes = {
 };
 
 export type Actions = {
-  delete_self: boolean;
```
Comment (mzhong9723) [inline packages/types/src/userSettings.ts:86]: Is this intentional?

File: packages/types/src/userSettings.ts [hunk 3]
```diff
@@ -101,7 +102,6 @@ export interface UserSettingsJSON extends ClerkResourceJSON {
   sign_in: SignInData;
   sign_up: SignUpData;
   password_settings: PasswordSettingsData;
-  actions: Actions;
```
Comment (jescalan) [inline packages/types/src/userSettings.ts:104]: This is just a move to make the placement consistent between the resource and json types

File: packages/types/src/userSettings.ts (hunk 2 above)
Comment (jescalan) [reply packages/types/src/userSettings.ts]: It was not, but tests failed because of it, and fixed now!

**Review by SokratisVidros — APPROVED** [review]

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**